adr/0002-resolve-schema-references-lazily.md

```diff
@@ -0,0 +1,19 @@
+# 3. Resolve schema references lazily
+
+Date: 2019-08-08
+
+## Status
+
+Accepted
+
+## Context
+
+We need
```

Comment:
We need to have minimal performance overhead on a test collection and avoid resolving that involves network requests